data/sql/updates/pending_db_world/rev_1554143974096714000.sql

```diff
@@ -0,0 +1,4 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1554143974096714000');
+
+REPLACE INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES (7572, 0, 0, 2, 62, 0, 100, 0, 842, 0, 0, 0, 0, 26, 2784, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Fallen Hero of the Horde - On Gossip Option 0 Selected - Quest Credit \'Fall From Grace\'');
+REPLACE INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES (7572, 0, 1, 3, 62, 0, 100, 0, 881, 0, 0, 0, 0, 26, 2801, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Fallen Hero of the Horde - On Gossip Option 1 Selected - Quest Credit \'A Tale of Sorrow\'');
```

Comment:
please use `DELETE` and `INSERT` instead of `REPLACE`

also don't forget the empty line at the end of the file

Reply:
depends on the change, sometiems replace into is better

In that case, do we need to delete and reinset the whole smart_scripts ? is the code above the whole smart_script for that entry or are there other rows that are untouched ?

Reply:
- we only have to `DELETE` the rows that we are going to `INSERT`. For consistency, we always have been using DELETE/INSERT instead of REPLACE since MaNGOS UDB times. One of the reasons is, it is more intuitive to understand what's going for people who are not familiar to SQL

- whenever we need to **edit** an existing row (without deleting and re-inserting it) then `UPDATE` shall be used